mustache.go

```diff
@@ -31,19 +31,31 @@ func (es ErrorSlice) Error() string {
 	return b.String()
 }
 
+func tryAddNode(visitedNodes map[node]struct{}, newNode node) error {
```

Comment:
possible nit: maybe change the name to `trackLookup` or `visitLookup`. The error here is user facing and the "node" concept is a bit opaque. Mustache calls this whole business "lookup".

Reply:
sounds good!